swift-sdk/Internal/InAppManager.swift

```diff
@@ -286,18 +302,21 @@ extension InAppManager : InAppSynchronizerDelegate {
 
         updateQueue.async {
             // Remove messages that are no present in server
-            self.removeDeletedMessages(messagesFromServer: messages)
+            let deletedInboxCount = self.removeDeletedMessages(messagesFromServer: messages)
             
             // add new ones
-            self.addNewMessages(messagesFromServer: messages)
+            let addedInboxCount = self.addNewMessages(messagesFromServer: messages)
             
             self.persister.persist(self.messagesMap.values)
             
-            // process inbox messages
-            self.processInboxMessages()
+            if deletedInboxCount + addedInboxCount > 0 {
```

Comment:
Shouldn't we trigger `onInboxChanged` if the number of added and deleted messages is the same? (deleted one message, added one message)

Reply:
Not sure I understand. This will be triggered if any message was added or deleted. So if you delete one message and add another. deletedCount = 1, and addedCount = 1.

Reply:
Oh, you're right. I thought the number of deleted entries was negative.